src/main/kotlin/com/atlassian/performance/tools/report/jstat/JstatConverter.kt

```diff
@@ -15,7 +15,10 @@ internal class JstatConverter {
     ): Path {
         val jstatCsvName = "${jstat.fileName.toString().substringBeforeLast(".")}.csv"
         val jstatCsv = jstat.parent.resolve(jstatCsvName)
-        jstatCsv.toFile().createNewFile()
+        jstatCsv.toFile().let {
+            it.delete()
```

Comment:
This returns a `Boolean` and doesn't throw on failures.
Using `Files.delete(jstatCsv)` would fix that.

Reply:
After making the change, I see why it fails on Windows:
```
java.nio.file.NoSuchFileException: C:\Users\Maciej\AppData\Local\Temp\junit17669322309799385493\jstat.csv
```

Debugging why this is thrown.

Reply:
You're right. file.delete() doesn't work here.

Reply:
Ofc, the first time around the file doesn't exist. After adding and `if exists`, I'm getting:
```
java.nio.file.FileSystemException: C:\Users\Maciej\AppData\Local\Temp\junit6943909163504257389\jstat.csv: The process cannot access the file because it is being used by another process.
```
Perhaps we're not releasing the file handle properly.

Reply:
We can't use `Files.delete` the file may not exist and it shouldn't throw an exception.

Reply:
Yep, the test is bugged, this is the fix:
```kotlin
jstatConverter.convertToCsv(logFile.toPath()).toFile().bufferedReader().use { it.readText() }
```
Note the `use`

Reply:
> 
> 
> We can't use `Files.delete` the file may not exist and it shouldn't throw an exception.

```kotlin
        if (jstatCsv.toFile().exists()) {
            Files.delete(jstatCsv)
        }
```

Reply:
Great work!

Reply:
It would be great to fail on compile time with not closed AutoCloseable :)

Reply:
Oh yes, I'd love that.

Reply:
There's an inspection for Java. But it's off by default. 

<img width="1001" alt="Screenshot 2019-04-18 at 13 22 55" src="https://user-images.githubusercontent.com/3447570/56357881-3ff1f900-61dd-11e9-8e71-64c71334238d.png">

Reply:
There's an ["close streams" check](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html?highlight=resources#os-method-may-fail-to-close-stream-os-open-stream) (not generic closeable), enforceable via [a Gradle plugin](https://plugins.gradle.org/plugin/com.github.spotbugs).

Reply:
Does it work with Kotlin?

Reply:
> Performs quality checks on your project's Java source files

Probably not 😞

Reply:
https://github.com/spotbugs/spotbugs/issues/573

Reply:
https://github.com/spotbugs/spotbugs/issues/573#issuecomment-447185559
Maybe it does work on bytecode level

Reply:
> FYI, there is a tool that may help to analyse Kotlin files:
https://github.com/arturbosch/detekt

It doesn't check Autocloseable but has other useful checks. And it can be extended by implementing own rules.